src/fromager/context.py

```diff
@@ -81,9 +77,19 @@ def pip_wheel_server_args(self) -> list[str]:
 
     @property
     def pip_constraint_args(self) -> list[str]:
-        if not self.input_constraints_file:
+        if not self.input_constraints_uri:
             return []
-        return ["--constraint", os.fspath(self.input_constraints_file)]
+
+        if self.input_constraints_uri.startswith(("https://", "http://", "file://")):
```

Comment:
This is going to cause `file://` URLs to be copied into the local temporary file. Those could be handled like regular files if we parsed the URL and extracted the path.

OTOH, if requests can already deal with that type of URL, then maybe it's OK.

Reply:
Actually it looks like requests can't handle `file://` url. If that is the case then should we even allow such URLs? @tiran 

```
>>> requests.get("file:///path/to/file")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/projects/fromager/venv/lib/python3.11/site-packages/requests/api.py", line 73, in get
    return request("get", url, params=params, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/projects/fromager/venv/lib/python3.11/site-packages/requests/api.py", line 59, in request
    return session.request(method=method, url=url, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/projects/fromager/venv/lib/python3.11/site-packages/requests/sessions.py", line 589, in request
    resp = self.send(prep, **send_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/projects/fromager/venv/lib/python3.11/site-packages/requests/sessions.py", line 697, in send
    adapter = self.get_adapter(url=request.url)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/projects/fromager/venv/lib/python3.11/site-packages/requests/sessions.py", line 792, in get_adapter
    raise InvalidSchema(f"No connection adapters were found for {url!r}")
requests.exceptions.InvalidSchema: No connection adapters were found for 'file:///path/to/file'
```

Reply:
It wasn't too bad to implement so lets keep it now :shrug:

Reply:
OK, what you have looks fine.